src/public/modules/forms/helpers/response-factory.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  ArrayResponse,
+  DisplayedResponseWithoutAnswer,
+  NestedResponse,
+  SingleResponse,
+} from '../../../../types/response'
+
+import {
+  ArrayAnswerResponse,
+  ErrorResponse,
+  Response,
+  SingleAnswerResponse,
+  TableResponse,
+} from './csv-response-classes'
+
+export const getResponseInstance = (
+  fieldRecordData: DisplayedResponseWithoutAnswer,
+): Response => {
+  switch (fieldRecordData.fieldType) {
+    case 'table':
+      if (isNestedResponse(fieldRecordData)) {
+        return new TableResponse(fieldRecordData)
+      }
+      break
+    case 'checkbox':
+      if (isArrayResponse(fieldRecordData)) {
+        return new ArrayAnswerResponse(fieldRecordData)
+      }
+      break
```

Comment:
this mixing of `switch` and `if` is a little confusing. I'd actually much rather stick to just if-else.